app/services/api/v3/concerns/filter_training_status.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Api::V3::Concerns::FilterTrainingStatus
+  extend ActiveSupport::Concern
+
+protected
+
+  def filter
+    params[:filter] ||= {}
+  end
+
+  def training_status_filter
+    filter[:training_status].to_s
+  end
+
+  def training_status
+    return if training_status_filter.blank?
+
+    unless training_status_filter.in?(valid_training_status)
+      raise Api::Errors::InvalidTrainingStatusError, I18n.t(:invalid_training_status, valid_training_status:)
```

Comment:
I think we'll need to rescue from this error in the `api_controller` as otherwise the error will come back as a 500 in the API, could you also add a test for this in the request specs?